examples/cd_text_amazon.ipynb

```diff
@@ -0,0 +1,553 @@
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "H0\n",
+      "Drift? No!\n",
+      "p-value: 0.898\n",
+      "\n",
+      "H1\n",
+      "Drift? Yes!\n",
+      "p-value: 0.000\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "make_predictions(ref_size=1000, test_size=1000, n_sample=2, stars_h1=4)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Classifier drift detector\n",
+    "\n",
+    "Now we will use the [ClassifierDrift detector](https://docs.seldon.io/projects/alibi-detect/en/latest/methods/classifierdrift.html) which uses a binary classification model to try and distinguish the reference from the test (H0 or H1) data. Drift is then detected on the difference between the prediction distributions on out-of-fold reference vs. test instances using a Kolmogorov-Smirnov 2 sample test on the prediction probabilities or via a binomial test on the binarized predictions. We use a pretrained transformer model but freeze its weights and only train the head which consists of 2 dense layers with a leaky ReLU non-linearity:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch.nn as nn\n",
+    "from transformers import DistilBertModel\n",
+    "\n",
+    "model_name = 'distilbert-base-uncased'\n",
+    "\n",
+    "class Classifier(nn.Module):\n",
+    "    def __init__(self) -> None:\n",
+    "        super().__init__()\n",
+    "        self.lm = DistilBertModel.from_pretrained(model_name)\n",
+    "        for param in self.lm.parameters():  # freeze language model weights\n",
+    "            param.requires_grad = False\n",
+    "        self.head = nn.Sequential(nn.Linear(768, 512), nn.LeakyReLU(.1), nn.Linear(512, 2))\n",
+    "    \n",
+    "    def forward(self, tokens) -> torch.Tensor:\n",
+    "        h = self.lm(**tokens).last_hidden_state\n",
+    "        h = nn.MaxPool1d(kernel_size=100)(h.permute(0, 2, 1)).squeeze(-1)\n",
+    "        return self.head(h)\n",
+    "\n",
+    "tokenizer = AutoTokenizer.from_pretrained(model_name)\n",
+    "model = Classifier()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from alibi_detect.cd import ClassifierDrift\n",
+    "from alibi_detect.utils.prediction import tokenize_transformer\n",
+    "\n",
+    "\n",
+    "def make_predictions(model, backend: str, ref_size: int, test_size: int, n_sample: int, stars_h1: int = 4) -> None:\n",
+    "    \"\"\" Create drift Classifier detector, init, sample data and make predictions. \"\"\"\n",
+    "    \n",
+    "    # batch_fn tokenizes each batch of instances of the reference and test set during training\n",
+    "    b = 'pt' if backend == 'pytorch' else 'tf'\n",
+    "    batch_fn = partial(tokenize_transformer, tokenizer=tokenizer, max_len=max_len, backend=b)\n",
```

Comment:
Can we always do this and get rid of the `preprocess_drift` function?

Reply:
`batch_fn` is used for `preprocess_batch_fn` here is different than the function of `preprocess_drift` which is a utility function for `preprocess_fn`.

Reply:
Again part of #298